worker/task.go

```diff
@@ -79,10 +82,41 @@ func NewTaskRun(
 		controller:    ctxtctl,
 		engine:        engine,
 		pluginManager: pluginManager,
+		queueURL:      config.Taskcluster.Queue.URL,
+		done:          make(chan struct{}),
 	}
 
+	go tr.reclaim(time.Time(claim.taskClaim.TakenUntil), tr.done)
+
 	return tr, nil
+}
+
+func (t *TaskRun) reclaim(until time.Time, done <-chan struct{}) {
+	duration := until.Sub(time.Now()).Seconds()
+	// Using a reclaim divisor of 1.3 with the default reclaim deadline (20 minutes),
+	// means that a reclaim event will happen with a few minutes left of the origin claim.
+	nextReclaim := duration / 1.3
+	select {
+	case <-time.After(time.Duration(nextReclaim * 1e+9)):
+		client := t.controller.Queue()
+		claim, err := reclaimTask(client, t.TaskID, t.RunID, t.log)
+		if err != nil {
+			t.log.WithField("error", err.Error()).Warn("Error reclaiming task")
```

Comment:
You're very right sir, I'll change that.  And with you pointing that out and me checking out the logrus documentation, I have discovered:
https://github.com/Sirupsen/logrus/blob/master/entry.go#L60

which I pretty much do in a lot of places so that can be simplified.

Thanks!